### DISCUSSION TITLE
NextJS: [zustand devtools middleware] Please install/enable Redux devtools extension

### DISCUSSION THREAD

**Discussion by adrenaline681** [pmndrs/zustand#1658] [General]
I have Redux devtools installed but I get the following error:

`[zustand devtools middleware] Please install/enable Redux devtools extension`

Im using NextJS and this warning appears on the server side only.

**Comment by dai-shi** [comment]
I think it fails to detect if it's on browser?
Can anyone help investigating the problem?

For now, you can pass `enabled: false` to avoid the warning.

**Reply by adrenaline681** [reply]
Thanks for the pointer, i was able to disable the devtools during server side rendering by adding:

`enabled: typeof window !== 'undefined' `

using typeof window !== 'undefined' is the way in Nextjs to check if you are on server side or client side

**Comment by dai-shi** [comment]
On second look, the warning is only shown if `enabled` is truthy.

https://github.com/pmndrs/zustand/blob/fbfcdc54e679cf1cb6d887078b4b9b19319417e9/src/middleware/devtools.ts#L219-L224

Do you specify `enabled` on your end?
